pkg/cloudprovider/providers/vsphere/vsphere_util.go

```diff
@@ -245,6 +249,102 @@ func getPbmCompatibleDatastore(ctx context.Context, dc *vclib.Datacenter, storag
 	return datastore, err
 }
 
+func getDatastoresForZone(ctx context.Context, dc *vclib.Datacenter, nodeManager *NodeManager, selectedZones []string) ([]*vclib.DatastoreInfo, error) {
+
+	var sharedDatastores []*vclib.DatastoreInfo
+
+	// Create cloudprovider.Zone type from selectedZones
+	var cpZones []cloudprovider.Zone
```

Comment:
This loop looks unnecessary and could be combined with the next loop.

```go
for _, zone := range selectedZones {
    var cpZone cloudprovider.Zone
    cpZone.FailureDomain = zone
    var sharedDatastoresPerZone []*vclib.DatastoreInfo
    hosts, err := nodeManager.GetHostsInZone(ctx, cpZone)
    if err != nil {
	return nil, err
    }
...
}
```

Reply:
Thanks for pointing out. Fixed.